x-pack/plugins/apm/public/components/alerting/ErrorCountAlertTrigger/index.tsx

```diff
@@ -51,45 +48,20 @@ export function ErrorRateAlertTrigger(props: Props) {
     ...alertParams,
   };
 
-  const threshold = isFinite(params.threshold) ? params.threshold : '';
-
   const fields = [
-    <PopoverExpression
-      value={getEnvironmentLabel(params.environment)}
-      title={i18n.translate('xpack.apm.errorRateAlertTrigger.environment', {
-        defaultMessage: 'Environment',
-      })}
-    >
-      <EuiSelect
-        value={params.environment}
-        options={environmentOptions}
-        onChange={(e) =>
-          setAlertParams(
-            'environment',
-            e.target.value as ErrorRateAlertTriggerParams['environment']
-          )
-        }
-        compressed
-      />
-    </PopoverExpression>,
-    <PopoverExpression
-      title={i18n.translate('xpack.apm.errorRateAlertTrigger.isAbove', {
-        defaultMessage: 'is above',
+    <ServiceField value={serviceName} />,
+    <EnvironmentField
+      currentValue={params.environment}
+      options={environmentOptions}
+      onChange={(e) => setAlertParams('environment', e.target.value)}
+    />,
+    <IsAboveField
+      value={params.threshold}
+      unit={i18n.translate('xpack.apm.errorCountAlertTrigger.errors', {
+        defaultMessage: ' errors',
```

Comment:
Extracting commonly-used field types into separate components. I was a little torn whether this improves readability by hiding boilerplate or hurts readability by adding an additional abstraction.
In the end I decided it was better.

Reply:
You are right. Was hoping you wouldn't notice 🙈
I'll fix it in a follow-up. For now I'll merge to avoid additional conflicts.